File: 49. Group Anagrams/GroupAnagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        if (strs == null || strs.length == 0) {
            return new ArrayList<>();
        }

        Map<String, List<String>> map = new HashMap<>();

        for (String s : strs) {
            char[] chars = s.toCharArray();
            Arrays.sort(chars);
            String sorted = new String(chars);
            map.computeIfAbsent(sorted, k -> new ArrayList<>()).add(s);
        }

        return new ArrayList<>(map.values());
    }
}

File: 49. Group Anagrams/GroupAnagrams.py
from collections import defaultdict

class Solution:
    def groupAnagrams(self, strs):
        if not strs:
            return []
        
        anagram_map = defaultdict(list)
        
        for s in strs:
            # Sắp xếp các ký tự của chuỗi và dùng nó làm key
            sorted_str = ''.join(sorted(s))
            anagram_map[sorted_str].append(s)
        
        return list(anagram_map.values())